### ISSUE TITLE
🚀 [Feature] Simple Chat

### ISSUE THREAD

**Issue by Mastercuber** [Human-Connection/Human-Connection#38]
**Describe the solution you'd like**
The idea is to use channels provided by the [mubsub](https://github.com/scttnlsn/mubsub) which uses mongodb as a hub and are synchronized with [feathers-sync](https://github.com/feathersjs-ecosystem/feathers-sync) to provide scalability, to implement a basic chat funtionality. User to user chat and group chat should be the first step.

**Describe alternatives you've considered**

**Additional context**
We could create a `chatrooms` and `chats` channel. The name of a `chatroom` could be the name of a event. To send data to a chatroom, just publish a event (message) with the name of the chatroom in the appropriate channel `chatrooms`. All client will receive the message and can add it to the view or update the history property etc...

**Comment by mimicc83** [comment]
Thanks for the summary of this feature. Please consider:

- using redis pub/sub instead as it's way faster on iops: https://redis.io/topics/pubsub
- 1:1 Chats (these should be protected so that only 2 users can enter)
- n:m Chats (protected and public)
- moderation/report function for fraudulent messages/harassment
- message deletion by users
- leaving chatrooms
- blacklist feature (needs to reflected when building this)
.... anything I missed!?

Can you @Mastercuber  or someone else please do a small pro/con research mubsub vs redis pubsub? 

Cheers

**Comment by mimicc83** [comment]
@Mastercuber please check this integration which works without any pubsub feature: https://github.com/feathers-plus/feathers-chat-vuex this way we remove reliance from database/storage level

**Comment by td-kdwelt** [comment]
Meanwhile, we should remove the "coming soon..." button. 
Why? people DO h8-2-w8 & "promises, promises"

**Comment by roschaefer** [comment]
@td-kdwelt could you send us a screenshot please?
@td-kdwelt is the "coming soon" button in the alpha version code base or the new "nitro" code base?

What does `"h8-2-w8" mean? If it means "hate to wait" I would suggest to avoid these kind of abbreviations. Consider that people of the open source community feel uneasy to tackle this issue just because they don't know what the participants of this chat are talking about :slightly_smiling_face: we want to encourage people to work independently on issues here. So we should use simple English which is easy to understand for everybody.

Also regarding the coming soon button. I found a "coming soon" message when you click on `versus` or `let's talk` in the current alpha version. 
![download 8](https://user-images.githubusercontent.com/2110676/50227512-2f07b700-03a6-11e9-87ae-e11d98a4c7e2.png)

I agree that a coming soon message is not very helpful. However, since we are an open-source project and we rely on volunteers, I like to ask people to work on the code. Similar to:
![screenshot 15](https://user-images.githubusercontent.com/2110676/50227629-72fabc00-03a6-11e9-9481-b66ec765c5ec.png)

We could ask "Hey do you want to help us with that?" and then show our email address and discord server and Github repository.

@td-kdwelt what's your opinion on it?

**Comment by appinteractive** [comment]
I love that idea! 💡

**Comment by td-kdwelt** [comment]
Sorry, haven´t been around for a couple of hours :)

>>roschaefer: I like to ask people to work on the code. 

 :) ask _me_ ..:)
Do you expect literals only or should I modify the script code? See BTW below.

Q: @td-kdwelt what's your opinion on it? 
A:  Your proposal: (s.th. like..)
"Hooray! You found a missing feature.
This is free and open-source software - it depends on your contribution!"

No "hoorays". Hoora for a bug? No way. 
Some dozens of users would tell you their "success" and, of course, they´d expect to get an answer.
There are things by far more important than writing mails.
Just remove/comment out all parts of the code that bear the property "not yet implemented"

Why? There is enough to repair, avoid, fix, or whatever you may call it the flaws the Alpha has.
I´m quite sure you are not aware of all of them. Nor am I. 
IMO, it would take quite a few days to doc them according to your "bug report templates" I´ve seen on your Github repo.
Another question:
Would you mind if I´d upload screenshots or other graphic stuff right here at zenhub?

BTW (see above):  
Is there a _written_ step by step documentation how to set up 

	the minimal OS/Version (I know how to install, but do we need extensions etc?)
	the needed additional software 
	environment settings
	_exact_ locations of the softwares
	_exact_ locations of the sources

and, of course, instructions how to make all that run locally

If yes, I haven´t found it yet.
If _not_, IMO it is time to start such a doc. If you start later, this task might be overwhelming.

That´s it for now
Bye

**Comment by td-kdwelt** [comment]
@appinteractive 
No help wanted anymore?

**Comment by appinteractive** [comment]
@td-kdwelt It is, only the descripton is from the old codebase, thats why I removed the label in the first place.

**Comment by td-kdwelt** [comment]
@appinteractive 

And what about the questions in the post _before_ ?
THX for our answer

The Duke

**Comment by roschaefer** [comment]
@td-kdwelt yes, go ahead, please create feature requests for every missing feature. Please use the search first, so we don't have duplicates. Keep in mind, we're focusing all the development effort on the new code base. Ie. if a feature is missing in the current alpha, it won't be implemented on the previous code base of the alpha. It will be implemented in the new code base (repos start with `Nitro-`).

Also regarding the "Hooray", it is intentionally positive. To find a bug or a missing feature is an accomplishment. It's also highlighting the fact that we're all in this together, we're not a private company that's running some business with the code or the user data. Everyone who sees a missing feature is welcome to contribute to the code base and implement missing features. That's development of free and open source software - run by the community and not by a company.

**Comment by td-kdwelt** [comment]
Just another Question

Is the pre-alpha-staging-site-nitro aka "Christmas Gift" the only one to "test" the current stage of the development?

How comes nobody found the "NULL-assignment" ? (wrong expression, but the best to describe it):
![image.png](https://images.zenhubusercontent.com/5c1a2dc59be21e037cb696a7/09be424d-6f17-4ea9-8e40-81632f354462)

Bye
the Duke

**Comment by appinteractive** [comment]
Hey the login issue is handled here but not deployed right now. https://github.com/Human-Connection/Nitro-Backend/issues/49

**Comment by appinteractive** [comment]
@td-kdwelt for non chat related questions, please open new issues so we do not chat here at the Ticket wich will make it unreadable very quickly. Thanks. 
Did you already had a look into our dokumentation? https://docs.human-connection.org/nitro

If something is still missing, open a separate ticket in the nitro-documentation project.

**Comment by td-kdwelt** [comment]
@appinteractive 
1. "for non chat related questions..."
2. " dokumentation"

got it. Thank you.

**Comment by mattwr18** [comment]
we still want a chat

**Comment by Tirokk** [comment]
Still valid …
